serverless/lambda/src/main/java/uk/gov/di/helpers/CookieHelper.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.di.helpers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CookieHelper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CookieHelper.class);
+
+    public static final String REQUEST_COOKIE_HEADER = "Cookie";
+
+    public static Optional<SessionCookieIds> parseSessionCookie(Map<String, String> headers) {
+        if (headers == null
+                || headers.isEmpty()
```

Comment:
We could probably change this check to just a null check, then move line 27 (`headers.getOrDefault...`) up here and have a second conditional for `cookies.isEmpty()/isBlank()`

The `getOrDefault` will step around the need for checking an empty collection, checking for the key presence, etc.

Reply:
I've refactored this class to use the HttpCookie class to parse the cookie instead of Regex.